File: buildSrc/src/main/groovy/com/rong360/plugin/ModuleAnnotationVisitor.java
package com.rong360.plugin;

import java.util.HashMap;

import org.objectweb.asm.AnnotationVisitor;

public class ModuleAnnotationVisitor extends AnnotationVisitor {

    private String nameString;
    private HashMap<String, String> valueMap;

    ModuleAnnotationVisitor(int api, AnnotationVisitor av, String name, HashMap<String, String> serviceMap) {
        super(api, av);
        nameString = name;
        valueMap = serviceMap;
    }

    @Override
    public void visit(String name, Object value) {
        super.visit(name, value);
        if ("register".equals(name)) {
            valueMap.put((String) value, nameString);
        } else if ("desc".equals(name)) {
            //todo show desc
        }
    }
}

File: msm-gradle-plugin/src/main/java/com/rong360/msm/plugin/ModuleAnnotationVisitor.kt
package com.rong360.msm.plugin

import java.util.HashMap

import org.objectweb.asm.AnnotationVisitor

class ModuleAnnotationVisitor internal constructor(api: Int, av: AnnotationVisitor, private val nameString: String?,
                                                   private val valueMap: HashMap<String, String>) : AnnotationVisitor(api, av) {

    override fun visit(name: String, value: Any) {
        super.visit(name, value)
        if ("register" == name) {
            nameString?.let {
                valueMap[value as String] = it
            }
        } else if ("desc" == name) {
            //todo show desc
        }
    }
}
